internal/utils/utils.go

```diff
@@ -5,19 +5,17 @@ import (
 	"context"
 	"database/sql"
 	"fmt"
-	"github.com/icinga/icingadb/pkg/driver"
-	"github.com/icinga/icingadb/pkg/icingadb"
-	"github.com/icinga/icingadb/pkg/types"
-	"github.com/icinga/icingadb/pkg/utils"
+	"github.com/icinga/icinga-go-library/database"
+	"github.com/icinga/icinga-go-library/types"
 	"github.com/jmoiron/sqlx"
 	"github.com/pkg/errors"
 	"slices"
 	"strings"
 )
 
 // BuildInsertStmtWithout builds an insert stmt without the provided column.
-func BuildInsertStmtWithout(db *icingadb.DB, into interface{}, withoutColumn string) string {
-	columns := db.BuildColumns(into)
+func BuildInsertStmtWithout(db *database.DB, into interface{}, withoutColumn string) string {
+	columns := slices.Clone(db.BuildColumns(into))
```

Comment:
I think it might be a good idea to do the clone within the library, i.e. never expose the internal slice from the cache so that there's no way to break it from the outside.